### PR TITLE
EdDSA signature creation via streaming

### LINKED ISSUE #137: Add EdDSA signature creation via streaming for arbitrary size messages
Certain blockchains (like Pocket) require signature of the entire message payload ( instead of the hash of the message). Doing such a signature generation for an arbitrary size message is not possible on the Ledger devices due to limited stack space. 

The only way to create signatures in limited stack space is by incrementally computing the SHA512 of the message.
Ref: https://datatracker.ietf.org/doc/html/rfc8032#page-13

The Ledger SDK must provide a way to compute EdDSA signature by consuming the message data incrementally, similar to the `Hash` interface.

A reference implementation is here https://github.com/alamgu/ledger-crypto-helpers/blob/8f46275fc7fe9b8169b4f689ce2a203fb02c993d/src/ed25519.rs#L72C6-L72C13

### PR DESCRIPTION
**Pull request by dfordivam**
Fixes #137

### CODE REVIEW

File: ledger_device_sdk/src/hash.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ pub trait HashInit: Sized {
             Ok(())
         }
     }
-    fn finalize(mut self, output: &mut [u8]) -> Result<(), HashError> {
+    fn finalize(&mut self, output: &mut [u8]) -> Result<(), HashError> {
         let output_size = self.get_size();
```
Comment (yogh333) [inline ledger_device_sdk/src/hash.rs:71]: I do not think this is required:when a hash is finalized, then its context shall be consumed. To compute a new hash, then `HashInit::new()` shall be called. It is the same behaviour as with `HashInit::hash()`

File: ledger_device_sdk/src/hash.rs [hunk 2]
```diff
@@ -67,7 +67,7 @@ pub trait HashInit: Sized {
             Ok(())
         }
     }
-    fn finalize(mut self, output: &mut [u8]) -> Result<(), HashError> {
+    fn finalize(&mut self, output: &mut [u8]) -> Result<(), HashError> {
```
Comment (yogh333) [inline ledger_device_sdk/src/hash.rs:70]: I do not think this is required:when a hash is finalized, then its context shall be consumed. To compute a new hash, then HashInit::new() shall be called. It is the same behaviour as with HashInit::hash()

**Comment by yogh333** [comment]
Hi, please check #144 as a new proposal

File: ledger_device_sdk/src/hash.rs (hunk 2 above)
Comment (dfordivam) [reply ledger_device_sdk/src/hash.rs]: Hi, I have removed this commit, now the PR uses `core::mem::take`

**Comment by yogh333** [comment]
Closed by #144

**Pull request closed without merging**